src/super_gradients/training/transforms/utils.py

```diff
@@ -89,21 +91,45 @@ def _get_bottom_right_padding_coordinates(input_shape: Tuple[int, int], output_s
     return PaddingCoordinates(top=0, bottom=pad_height, left=0, right=pad_width)
 
 
-def _pad_image(image: np.ndarray, padding_coordinates: PaddingCoordinates, pad_value: int) -> np.ndarray:
+def _pad_image(image: np.ndarray, padding_coordinates: PaddingCoordinates, pad_value: Union[int, Tuple[int, ...]]) -> np.ndarray:
     """Pad an image.
 
     :param image:       Image to shift. (H, W, C) or (H, W).
     :param pad_h:       Tuple of (padding_top, padding_bottom).
     :param pad_w:       Tuple of (padding_left, padding_right).
-    :param pad_value:   Padding value
+    :param pad_value:   Padding value. Can be a single scalar (Same value for all channels) or a tuple of values.
+                        In the latter case, the tuple length must be equal to the number of channels.
     :return:            Image shifted according to padding coordinates.
     """
     pad_h = (padding_coordinates.top, padding_coordinates.bottom)
     pad_w = (padding_coordinates.left, padding_coordinates.right)
 
     if len(image.shape) == 3:
-        return np.pad(image, (pad_h, pad_w, (0, 0)), "constant", constant_values=pad_value)
+        _, _, num_channels = image.shape
+
+        if isinstance(pad_value, numbers.Number):
+            pad_value = tuple([pad_value] * num_channels)
+        else:
+            if isinstance(pad_value, typing.Sized) and len(pad_value) != num_channels:
+                raise ValueError(f"A pad_value tuple ({pad_value} length should be {num_channels} for an image with {num_channels} channels")
+
+            pad_value = tuple(pad_value)
+
+        padded_channels = []
+        for channel_index, pad_value_channel in enumerate(pad_value):
```

Comment:
You can actually replace all of this new code with a single line:
```
return np.pad(image, (pad_h, pad_w, (0, 0)), "constant", constant_values=((pad_value, pad_value), (pad_value, pad_value), (0, 0)))
```
You don't even need the `if`
and since this function is going to be called many time, it is better if it is vectorized

Reply:
In short, we can't. Believe me I tried. Maybe I missed something but no combination did not work.

We also can't use cv2.copyMakeBorder since it does not work correct with C>4 (rest of channels padded with 0 instead of pad values)

This works fine if your pad value is scalar. However I was not able to make it work when pad value is array of [C] elements (for example pad_value = (127,128,129).

Reply:
@ofrimasad thanks for the tip. Fixed in a382457af54cd0a765087032f7c7c8dcafe1f566